java/compaction/compaction-core/src/main/java/sleeper/compaction/core/job/dispatch/CompactionJobDispatcher.java

```diff
@@ -119,12 +123,26 @@ private void returnToQueueWithDelay(TableProperties tableProperties, CompactionJ
                 delaySeconds, request.getBatchKey());
     }
 
+    private void delete(String batchKey, int batchSize) {
+        LOGGER.info("Deleting batch after sending {} jobs", batchSize);
+        try {
+            deleteBatch.delete(instanceProperties.get(DATA_BUCKET), batchKey);
+        } catch (SdkClientException e) {
+            LOGGER.error("Batch not deleted", e);
```

Comment:
We could still do with unit testing this case. I think we can catch RuntimeException rather than any specific exception type.